include/boost/process/v2/impl/exe.ipp

```diff
@@ -0,0 +1,324 @@
+    int mib[4]; 
+    std::size_t len;
+    mib[0] = CTL_KERN;
+    mib[1] = KERN_PROC_ARGS;
+    mib[2] = pid;
+    mib[3] = KERN_PROC_PATHNAME;
+    if (sysctl(mib, 4, nullptr, &len, nullptr, 0) == 0)
+    {
+        std::string strbuff;
+        strbuff.resize(len, '\0');
+        char *exe = strbuff.data();
+        if (sysctl(mib, 4, exe, &len, nullptr, 0) == 0)
+        {
+            char buffer[PATH_MAX];
+            if (realpath(exe, buffer))
+            {
+                path = buffer;
+            }
+        }
+    }
+    if (path.empty())
+        ec = detail::get_last_error();
+    return path;
+}
+
+#elif defined(__OpenBSD__)
+
+filesystem::path exe_path(boost::process::v2::pid_type pid, std::error_code & ec)
+{
+    std::string path;
+    auto is_executable = [](boost::process::v2::pid_type pid, std::string in, std::string *out, std::error_code & ec) 
+    {
+        *out = "";
+        bool success = false;
+        struct stat st;
+        char executable[PATH_MAX];
+        if (!stat(in.c_str(), &st) && 
+            (st.st_mode & S_IXUSR) && 
+            (st.st_mode & S_IFREG) && 
+            realpath(in.c_str(), executable)) 
+        {
+            int cntp = 0;
+            kinfo_file *kif = nullptr;
+            kd = kvm_openfiles(nullptr, nullptr, nullptr, KVM_NO_FILES, nullptr);
+            if (!kd)
+            {
+                ec = detail::get_last_error();
+                return false;
+            }
+            if ((kif = kvm_getfiles(kd, KERN_FILE_BYPID, pid, sizeof(struct kinfo_file), &cntp))) 
+            {
+                for (int i = 0; i < cntp; i++) 
+                {
+                    if (kif[i].fd_fd == KERN_FILE_TEXT && 
+                        (st.st_dev == static_cast<dev_t>(kif[i].va_fsid) || 
+                        st.st_ino == static_cast<ino_t>(kif[i].va_fileid))) 
+                    {
+                        *out = executable;
+                        success = true;
+                        break;
+                    }
+                }
+            }
+            kvm_close(kd);
+        }
+        if (*out.empty())
+            ec = detail::get_last_error();
+        return success;
+    };
+    std::vector<std::string> buffer = cmdline_from_proc_id(pid, ec);
+    if (!buffer.empty()) 
+    {
+        bool is_exe = false;
+        std::string argv0;
+        if (!buffer[0].empty()) 
+        {
+            if (buffer[0][0] == '/') 
+            {
+                argv0 = buffer[0];
+                is_exe = is_executable(pid, argv0.c_str(), &path, ec);
+            } 
+            else if (buffer[0].find('/') == std::string::npos) 
+            {
+                std::string penv = envvar_value_from_proc_id(pid, "PATH");
+                if (!penv.empty()) {
+                    std::vector<std::string> env;
+                    std::string tmp;
+                    std::stringstream sstr(penv); 
+                    while (std::getline(sstr, tmp, ':')) 
```

Comment:
That won't do. I got a whole set of facilities in the `environment` header to do all this stuff without the overhead of `string`  and `stringstream`. Please look there and minimize the amount of allocations.